test/NameLookup/name_lookup.swift

```diff
@@ -288,23 +288,25 @@ class ThisDerived1 : ThisBase1 {
     self.Type // expected-error {{type 'ThisDerived1' has no member 'Type'}}
   }
 
+  // FIXME(SR-15250): Partial application diagnostic is applied incorrectly for some test cases.
+  // TODO(SR-15251): Requires SR-15250, Add a fixit for adding '()' for partial application.
```

Comment:
Sorry for the bikeshedding, but I think the `TODO` comment about adding fix-it could go where the diagnostic is emitted(emitDiagnostic call). Also, a little bit on the phrasing "TODO(diagnostics): SR-15250, Add a "did you mean to call it?" note with a fix-it for inserting '()' if function type has no params or all have a default value"

Reply:
yeah that makes sense. will move it, thanks!

Reply:
Done 👍🏿

Reply:
Thank you =]